Modules/Components/Niftyreg/include/selxDisplacementFieldNiftiToItkImageSinkComponent.h

```diff
@@ -0,0 +1,113 @@
+/*=========================================================================
+ *
+ *  Copyright Leiden University Medical Center, Erasmus University Medical
+ *  Center and contributors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#ifndef selxDisplacementFieldNiftiToItkImageSinkComponent_h
+#define selxDisplacementFieldNiftiToItkImageSinkComponent_h
+
+#include "selxSuperElastixComponent.h"
+#include "selxInterfaces.h"
+#include "selxNiftyregInterfaces.h"
+#include "selxNiftiToItkImage.h"
+#include "selxSinksAndSourcesInterfaces.h"
+#include "selxItkObjectInterfaces.h"
+
+#include <string.h>
+#include "itkImageFileWriter.h"
+#include "itkImportImageFilter.h"
+#include "selxAnyFileWriter.h"
+#include "selxFileWriterDecorator.h"
+
+namespace selx
+{
+template< int Dimensionality, class TPixel >
+class DisplacementFieldNiftiToItkImageSinkComponent :
+  public SuperElastixComponent<
+  Accepting< NiftyregDisplacementFieldImageInterface< TPixel >, itkImageDomainFixedInterface< Dimensionality >>,
+  Providing< SinkInterface, AfterRegistrationInterface >
+  >
+{
+public:
+
+  /** Standard ITK typedefs. */
+  typedef DisplacementFieldNiftiToItkImageSinkComponent< Dimensionality, TPixel > Self;
+  typedef SuperElastixComponent<
+    Accepting< NiftyregDisplacementFieldImageInterface< TPixel >, itkImageDomainFixedInterface< Dimensionality >>,
+    Providing< SinkInterface, AfterRegistrationInterface >
+    >                                            Superclass;
+  typedef std::shared_ptr< Self >       Pointer;
+  typedef std::shared_ptr< const Self > ConstPointer;
+
+  typedef itkImageDomainFixedInterface< Dimensionality > ItkImageDomainInterfaceType;
+  typedef std::shared_ptr< nifti_image >                 NiftiImagePointer;
+
+  typedef NiftyregDisplacementFieldImageInterface< TPixel >             NiftiDisplacementFieldInterfaceType;
+  typedef typename itk::Image< itk::Vector< TPixel, Dimensionality >, Dimensionality >    ItkDisplacementFieldType;
+  typedef typename ItkDisplacementFieldType::Pointer                                     ItkDisplacementFieldPointer;
+  typedef typename itk::ImageFileWriter< ItkDisplacementFieldType >                      ItkDisplacementFieldWriterType;
```

Comment:
Should not have 'typename' here: itk::ImageFileWriter<...> is always a type, by definition!